crates/project/src/terminals.rs

```diff
@@ -379,17 +369,60 @@ fn prepare_ssh_shell(
     // todo(windows)
     #[cfg(not(target_os = "windows"))]
     std::fs::set_permissions(ssh_path, smol::fs::unix::PermissionsExt::from_mode(0o755))?;
-    let path = format!(
-        "{}:{}",
-        tmp_dir.to_string_lossy(),
-        env.get("PATH")
-            .cloned()
-            .or(env::var("PATH").ok())
-            .unwrap_or_default()
-    );
-    env.insert("PATH".to_string(), path);
+
+    prepare_env_paths(env, tmp_dir)?;
 
     let mut args = shlex::split(&ssh_command).unwrap_or_default();
     let program = args.drain(0..1).next().unwrap_or("ssh".to_string());
     Ok(Shell::WithArguments { program, args })
 }
+
+fn prepare_env_paths(env: &mut HashMap<String, String>, add_path: &Path) -> anyhow::Result<()> {
+    let mut env_paths = vec![add_path.to_path_buf()];
+    if let Some(path) = env.get("PATH").or(env::var("PATH").ok().as_ref()) {
+        let mut paths = std::env::split_paths(&path).collect::<Vec<_>>();
+        env_paths.append(&mut paths);
+    }
+
+    let paths = std::env::join_paths(env_paths).context("failed to create PATH env variable")?;
+    env.insert("PATH".to_string(), paths.to_string_lossy().to_string());
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use collections::HashMap;
+
+    #[test]
+    fn test_prepare_env_paths() {
+        let tmp_path = std::path::PathBuf::from("/tmp/new");
+        let mut env = HashMap::default();
+        let old_path = if cfg!(windows) {
+            "/usr/bin;/usr/local/bin"
+        } else {
+            "/usr/bin:/usr/local/bin"
+        };
+        env.insert("PATH".to_string(), old_path.to_string());
+        env.insert("OTHER".to_string(), "aaa".to_string());
+
+        super::prepare_env_paths(&mut env, &tmp_path).unwrap();
+        if cfg!(windows) {
+            assert_eq!(env.get("PATH").unwrap(), &format!("/tmp/new;{}", old_path));
+        } else {
+            assert_eq!(env.get("PATH").unwrap(), &format!("/tmp/new:{}", old_path));
+        }
+        assert_eq!(env.get("OTHER").unwrap(), "aaa");
+
+        let mut env = HashMap::default();
+        env.insert("OTHER".to_string(), "aaa".to_string());
+        let os_path = std::env::var("PATH").unwrap();
+        super::prepare_env_paths(&mut env, &tmp_path).unwrap();
+        if cfg!(windows) {
+            assert_eq!(env.get("PATH").unwrap(), &format!("/tmp/new;{}", os_path));
+        } else {
+            assert_eq!(env.get("PATH").unwrap(), &format!("/tmp/new:{}", os_path));
+        }
+        assert_eq!(env.get("OTHER").unwrap(), "aaa");
+    }
```

Comment:
I'd split this out into a separate test so it's clear that one is the "empty" case and the other isn't.

Reply:
Done